File: 3055-maximum-odd-binary-number/maximum-odd-binary-number.cpp
class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int ans = 0 ; 
        for(int i = 0 ;i<s.length() ;i++){
            if(s[i] == '1') ans++;
        }
        int n = s.length();
        for(int i = 0 ;i<n-1; i++){
            if(ans>1){
                ans--;
                s[i] = '1';
            }
            else s[i] = '0';
        }
        s[n-1] = '1';
        return s;
    }
};

File: 3055-maximum-odd-binary-number/maximum-odd-binary-number.java
class Solution {
    public String maximumOddBinaryNumber(String s) {
        String ans = "";
        int cnt  = 0 ; 
        int n = s.length();
        for(int i = 0 ;i<n ;i++){
            if(s.charAt(i)=='1') cnt++;
        }
        for(int i = 0 ;i <n-1 ;i++){
            if(cnt>1){
                cnt--;
                ans = ans+'1';
            }
            else {
                ans = ans+'0';
            }
        }
        ans = ans+'1';
        return ans;
    }
}